WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -219,6 +236,16 @@ class PostCoordinator: NSObject {
         })
     }
 
+    private func uploadRevision(post: AbstractPost) {
+        guard post.hasRemote() else {
```

Comment:
On further testing, this makes sense for this scenario:

1. Go offline
1. Create a published post
2. Tap _Cancel_ in the post list
3. The post will be automatically uploaded as a draft

It makes sense. Or else, the post will be published. 

I am still not sure of the placement of the status change in here though. It can be really surprising when you call `PostCoordinator.autoSave()` and the `post.status` is changed under your nose. 

I wonder if we can change the post status when pressing _Cancel_ in the post list instead. 🤔  What do you think?

Reply:
More specifically, this part: 

https://github.com/wordpress-mobile/WordPress-iOS/blob/979540d22fa8d164f88d9f4917cc3031fedb1a2a/WordPress/Classes/Services/PostCoordinator.swift#L291-L305

It already makes changes to the `post` so I think it's a good location for it.

Reply:
I did some tests and I simplified it to:

```
    private func uploadRevision(post: AbstractPost) {
        post.status = .draft
        mainService.autoSave(post, success: { uploadedPost, _ in }, failure: { _ in })
    }
```

The problem here is that I need to change the status to `draft`, otherwise, it will never be saved (or uploaded, or anything). If we don't do this change, `autoSave` is called and the post is never saved.

Another option would be to set the status to `draft` when the user taps cancel, but I'm not sure about it.

Suggestions?

Reply:
> Another option would be to set the status to draft when the user taps cancel, but I'm not sure about it.

Yup. This is what I suggested in https://github.com/wordpress-mobile/WordPress-iOS/pull/12555#discussion_r330140896. And IIRC, this is how we implemented it on Android too. 

Resetting it during cancel also fixes other scenarios like this:

1. Create a draft and publish it while offline
2. Press Cancel in Post List
3. Go online and edit the draft. 
4. Press X → Update Post
5. ❗️The draft is automatically **published**.

Reply:
I tried this and it seems to be working well. 

```swift
func PostCoordinator.cancelAutoUploadOf(...) {
    ...
    if !post.hasRemote() {
         post.status = .draft
    }

    ...
}
```

Reply:
Done in 79991f5cc2d0620af5b9d2adf0a9c93c670663ac